tests/phpunit/integration/Core/Email_Reporting/Email_LogTest.php

```diff
@@ -0,0 +1,213 @@
+		wp_set_current_user( 0 );
+	}
+
+	public function test_registers_delivery_statuses() {
+		foreach (
+			array(
+				Email_Log::STATUS_SENT,
+				Email_Log::STATUS_FAILED,
+				Email_Log::STATUS_SCHEDULED,
+			) as $status
+		) {
+			$status_object = get_post_status_object( $status );
+			$this->assertNotNull( $status_object, sprintf( 'Status %s should be registered.', $status ) );
+
+			$this->assertFalse( $status_object->public, 'Status should not be public.' );
+			$this->assertTrue( $status_object->internal, 'Status should be internal.' );
+			$this->assertTrue( $status_object->exclude_from_search, 'Status should be excluded from search.' );
+			$this->assertFalse( $status_object->show_in_admin_all_list, 'Status should not show in admin all list.' );
+			$this->assertFalse( $status_object->show_in_admin_status_list, 'Status should not show in status list.' );
+			$this->assertIsArray( $status_object->label_count, 'Status label count should be registered.' );
+		}
+	}
+
+	public function test_registers_meta_keys() {
+		$meta = get_registered_meta_keys( 'post', Email_Log::POST_TYPE );
+
+		$expected_keys = array(
+			Email_Log::META_REPORT_FREQUENCY,
+			Email_Log::META_BATCH_ID,
+			Email_Log::META_SEND_ATTEMPTS,
+			Email_Log::META_ERROR_DETAILS,
+			Email_Log::META_REPORT_REFERENCE_DATES,
+		);
+
+		foreach ( $expected_keys as $key ) {
+			$this->assertArrayHasKey( $key, $meta, sprintf( 'Meta key %s should be registered.', $key ) );
+			$this->assertFalse( $meta[ $key ]['show_in_rest'], sprintf( 'Meta key %s should not show in REST.', $key ) );
+			$this->assertTrue( $meta[ $key ]['single'], sprintf( 'Meta key %s should be single.', $key ) );
+			$this->assertIsCallable( $meta[ $key ]['sanitize_callback'], sprintf( 'Meta key %s should have a sanitize callback.', $key ) );
+		}
+	}
+
+	public function test_meta_sanitization_behaviour() {
+		$meta = get_registered_meta_keys( 'post', Email_Log::POST_TYPE );
+
+		$frequency_callback = $meta[ Email_Log::META_REPORT_FREQUENCY ]['sanitize_callback'];
+		$this->assertSame( Email_Reporting_Settings::FREQUENCY_WEEKLY, call_user_func( $frequency_callback, 'Weekly' ), 'Sanitize frequency should accept weekly values.' );
+		$this->assertSame( '', call_user_func( $frequency_callback, 'yearly' ), 'Sanitize frequency should reject unsupported values.' );
+
+		$attempts_callback = $meta[ Email_Log::META_SEND_ATTEMPTS ]['sanitize_callback'];
+		$this->assertSame( 0, call_user_func( $attempts_callback, -1 ), 'Sanitize attempts should clamp negative numbers to zero.' );
+		$this->assertSame( 5, call_user_func( $attempts_callback, '5' ), 'Sanitize attempts should cast numeric strings.' );
+
+		$batch_id_callback = $meta[ Email_Log::META_BATCH_ID ]['sanitize_callback'];
+		$long_batch_id     = str_repeat( 'a', 300 );
+		$sanitized_batch   = call_user_func( $batch_id_callback, $long_batch_id );
+		$this->assertLessThanOrEqual( 191, strlen( $sanitized_batch ), 'Sanitize batch ID should enforce length limits.' );
```

Comment:
Please leave this "magic constant" value here in the tests though, since that way we'll be notified if we change/remove the length value 🙂